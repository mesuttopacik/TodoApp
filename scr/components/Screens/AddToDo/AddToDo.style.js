import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    flex: 1,
    // justifyContent:'flex-end',
    backgroundColor: '#FFFBE9',
  },
  addToDoWrapper: {
    flex: 1,
    position: 'absolute',
    bottom: 50,
    width: '100%',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-evenly',
  },

  textInput: {
    padding: 16,
    borderWidth: 1,
    width: '80%',
    borderColor: '#D18CE0',
    backgroundColor: '#FFFBE9',
    borderRadius: 55,
  },
});
