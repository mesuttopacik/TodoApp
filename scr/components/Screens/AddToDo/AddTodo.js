import React, {useState} from 'react';
import {Text, Platform, TextInput, TouchableOpacity, View} from 'react-native';
import styles from './AddToDo.style';
import ToDoCard from '../../ToDoCard';
import {useDispatch} from 'react-redux';
import Addbutton from '../../AddButton';

const AddToDo = () => {
  const [task, setTask] = useState();
  const [todos, setTodos] = useState([]);

  const dispatch = useDispatch();
  handleSetDoneList = doneTask => {
    dispatch({type: 'DONE_TASK', payload: {toDo: doneTask}});
  };
  const toDoHandler = () => {
    if (!task) {
      return;
    }
    setTodos([...todos, task]);
    setTask('');
  };

  onDeltask = index => {
    let todosCopy = [...todos];
    let doneTask = todosCopy.splice(index, 1);
    handleSetDoneList(doneTask[0]);
    setTodos(todosCopy);
  };
  return (
    <View
      enabled
      behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
      style={styles.container}>
      <ToDoCard todos={todos} onDeltask={onDeltask} />
      <View style={styles.addToDoWrapper}>
        <TextInput
        testID='todo-input'
        placeholder='Add..'
          style={styles.textInput}
          onChangeText={text => setTask(text)}
          value={task}
        />
        <Addbutton onClick={toDoHandler} />
      </View>
    </View>
  );
};

export default AddToDo;
