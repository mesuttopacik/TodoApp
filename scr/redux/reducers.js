export default function (state, action) {
  switch (action.type) {
    case 'DONE_TASK':
      const newDoneList = [...state.doneList, action.payload.toDo];
      return {...state, doneList: newDoneList};
      case 'REMOVE_DONE_TASK':
        let newList = [...state.doneList];
        let filtiredList = newList.filter((i, index) => index !== action.payload.index)
        return {...state, doneList: filtiredList};
    default:
        return state
  }
}
