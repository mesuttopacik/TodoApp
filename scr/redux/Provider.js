import React from 'react';

import {Provider} from 'react-redux';
import {createStore} from 'redux';
import reducers from './reducers';
import initialState from './store';

export const store = createStore(reducers, initialState);

const ToDoProvider = ({children}) => {
  return <Provider store={store}>{children}</Provider>;
};

export default ToDoProvider;