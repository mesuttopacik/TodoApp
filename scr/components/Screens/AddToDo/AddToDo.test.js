import {fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import AddToDo from './AddToDo';
import {Provider} from 'react-redux';
import {store} from '../../../redux/Provider';

describe('AddToDo tests', () => {
  let comp, input, submitButton, list;
  beforeEach(() => {
    comp = render(
      <Provider store={store}>
        <AddToDo />
      </Provider>,
    );
    input = comp.getByTestId('todo-input');
    submitButton = comp.getByTestId('button-toucable');
    list = comp.getByTestId('undone-task').props;
  });

  test('should match with snapshot', () => {
    expect(comp).toMatchSnapshot();
  });

  test('should add item to todolist if clicked AddButton', () => {
    fireEvent.changeText(input, 'do something');
    fireEvent.press(submitButton);
    list = comp.getByTestId('undone-task').props;
    expect(list.data.length).toBe(1);
  });

  test('should not add item to todolist if input is null AddButton', () => {
    fireEvent.press(submitButton);
    expect(list.data.length).toBe(0);
  });
});
