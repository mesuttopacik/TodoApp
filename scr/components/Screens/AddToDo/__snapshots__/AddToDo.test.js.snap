// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToDo tests should match with snapshot 1`] = `
<View
  behavior="padding"
  enabled={true}
  style={
    Object {
      "backgroundColor": "#FFFBE9",
      "flex": 1,
    }
  }
>
  <View>
    <RCTScrollView
      data={Array []}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      testID="undone-task"
      viewabilityConfigCallbackPairs={Array []}
    >
      <View />
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 50,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-evenly",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Add.."
      style={
        Object {
          "backgroundColor": "#FFFBE9",
          "borderColor": "#D18CE0",
          "borderRadius": 55,
          "borderWidth": 1,
          "padding": 16,
          "width": "80%",
        }
      }
      testID="todo-input"
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="button-toucable"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#D18CE0",
            "borderColor": "#D18CE0",
            "borderRadius": 25,
            "borderWidth": 1,
            "height": 50,
            "justifyContent": "center",
            "width": 50,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFBE9",
              "fontSize": 20,
            }
          }
        >
          +
        </Text>
      </View>
    </View>
  </View>
</View>
`;
